### PR TITLE
[SCB-950] modify consumer producer metrics and add edge metrics

### PR DESCRIPTION
**Pull request by heyile**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/statistics/MeterDetailStatisticsModel.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+import java.util.List;
+import java.util.Map;
+
+import org.apache.servicecomb.metrics.core.meter.invocation.MeterInvocationConst;
+import org.apache.servicecomb.metrics.core.publish.model.invocation.OperationPerf;
+import org.apache.servicecomb.metrics.core.publish.model.invocation.PerfInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MeterDetailStatisticsModel {
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/statistics/MeterDetailStatisticsModel.java]: why need this model?

enum the stages, just like the old logic, seems simpler?

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java [hunk 2]
```diff
@@ -106,83 +117,117 @@ protected void printThreadPoolMetrics(DefaultPublishModel model, StringBuilder s
...
   }
 
+  protected void printEdgeLog(DefaultPublishModel model, StringBuilder sb) {
+    OperationPerfGroups edgePerf = model.getEdge().getOperationPerfGroups();
+    if (edgePerf == null) {
+      return;
+    }
+    sb.append(String.format(SIMPLE_HEADER, "edge"));
+    //clear old data
+    MeterStatisticsManager.statisticsOperationMap.clear();
```
Comment (wujimin) [inline metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java]: why need global variable?

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/statistics/MeterDetailStatisticsModel.java (hunk 1 above)
Comment (heyile) [reply metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/statistics/MeterDetailStatisticsModel.java]: well. the data structure is different with  before.
for example:

old:
rest.200:
  xx  xxx xx xx  sayHi
  xxx xx xx xx   sayHello
rest.400:
  xx  xxx xx xx  sayHi
  xxx xx xx xx   sayHello

now: 
sayHi:
   rest200:
      xx xx xxx
   rest400:
      xx xx xxx
sayHello:
   rest200:
      xx xx xxx
   rest400:
      xx xx xxx
  Hence, I have to load all  data, And then restructure it

File: metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java (hunk 2 above)
Comment (heyile) [reply metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java]: I just use it to load data temporarily as I need restructure the data

Comment (heyile) [reply metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/publish/DefaultLogPublisher.java]: I have removed the global variable

**Pull request merged**